SU2_CFD/src/output/filewriter/CSTLFileWriter.cpp

```diff
@@ -0,0 +1,352 @@
+/*!
+ * \file CSTLFileWriter.cpp
+ * \brief STL Writer output class
+ * \author T. Kattmann, T. Albring
+ * \version 7.0.0 "Blackbird"
+ *
+ * SU2 Project Website: https://su2code.github.io
+ *
+ * The SU2 Project is maintained by the SU2 Foundation
+ * (http://su2foundation.org)
+ *
+ * Copyright 2012-2019, SU2 Contributors (cf. AUTHORS.md)
+ *
+ * SU2 is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 2.1 of the License, or (at your option) any later version.
+ *
+ * SU2 is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public
+ * License along with SU2. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "../../../include/output/filewriter/CSTLFileWriter.hpp"
+#include "../../../include/output/filewriter/CParallelDataSorter.hpp"
+#include <iomanip> /*--- Used for ofstream.precision(int). ---*/
+
+
+const string CSTLFileWriter::fileExt = ".stl";
+
+
+CSTLFileWriter::CSTLFileWriter(vector<string> fields,
+                               unsigned short nDim,
+                               string fileName,
+                               CParallelDataSorter *dataSorter) :
+                CFileWriter(std::move(fields),
+                            std::move(fileName),
+                            dataSorter,
+                            fileExt,
+                            nDim){}
+
+
+CSTLFileWriter::~CSTLFileWriter(){}
+
+
+void CSTLFileWriter::ReprocessElementConnectivity(){
+
+  /*--- Re-process the element connectivity information and store that appropriatly such that
+    this information can be accessed in step 2. It is copied from CTecplotBinaryFileWriter.cpp +162 and mildly modified. ---*/
+
+  unsigned long nParallel_Tria = dataSorter->GetnElem(TRIANGLE),
+                nParallel_Quad = dataSorter->GetnElem(QUADRILATERAL);
+  unsigned short iVar;
+
+  num_nodes_to_receive.resize(size, 0);
+  values_to_receive_displacements.resize(size);
+  halo_nodes.clear();
+  sorted_halo_nodes.clear();
+
+  /* We output a single, partitioned zone where each rank outputs one partition. */
+  vector<int> partition_owners;

```

Comment:
This variable is not used anymore

Reply:
removed together with all other occurences